internal/sidekick/rust/annotate.go

```diff
@@ -690,6 +701,119 @@ func (c *codec) addFeatureAnnotations(model *api.API, ann *modelAnnotations) {
 	}
 }
 
+// makeFieldAccessor generates the Rust accessor code for a field.
+// It handles optional fields and oneofs correctly.
+// parentAccessor is the accessor for the parent message (e.g. "req" or "s").
+func makeFieldAccessor(field *api.Field, parentAccessor string) string {
```

Comment:
I cannot grok how this works for fields nested multiple levels.  I am sure you thought of that, I can see there are test cases for it.  I am a bit tired, and maybe "future us" will be too.  Maybe a comment in the mustache template?